### PR TITLE
Exclude PInvokes declared on other modules. We don't yet encode crossmodule references

### PR DESCRIPTION
**Pull request by fadimounir**
This is because of this code in ZapSig::EncodeMethod: 
https://github.com/dotnet/coreclr/blob/cd418635d5db40df67ffb45de74f61ca8df88207/src/vm/zapsig.cpp#L1217

I'm not trying to fix the encoding here, just special case "extern" methods declared on other modules (to unblock WPF scenarios... PresentationCore.dll and PresentationFramework.dll have that.

I'll fix cross module reference encoding later when I get a chance

### CODE REVIEW

File: src/vm/dllimport.cpp [hunk 1]
```diff
@@ -3403,6 +3403,14 @@ BOOL NDirect::MarshalingRequired(MethodDesc *pMD, PCCOR_SIGNATURE pSig /*= NULL*
         if (pMD->IsNDirect() && ((NDirectMethodDesc *)pMD)->IsClassConstructorTriggeredByILStub())
             return TRUE;
 
+        if (IsReadyToRunCompilation() && !pMD->GetModule()->IsInCurrentVersionBubble())
```
Comment (jkotas) [inline src/vm/dllimport.cpp]: It may be better check this in`ZapInfo::pInvokeMarshalingRequired`

**Comment by jkotas** [comment]
LGTM otherwise

**Comment by jkotas** [comment]
Also, could you please add a regression test for this to the R2R unit test? https://github.com/dotnet/coreclr/blob/master/tests/src/readytorun/tests/main.cs

**Comment by fadimounir** [comment]
Hmm... Looks like the R2R unit test already covers this scenario in TestInterop(). Maybe this change fixes #23643

**Pull request merged**